migrations/20210927094004_payment/migration.sql

```diff
@@ -0,0 +1,283 @@
+CREATE TYPE "document_type" AS ENUM ('invoice', 'receipt', 'medical_record', 'other');
+
+-- CreateEnum
+CREATE TYPE "transaction_status" AS ENUM ('initial', 'invalid', 'incomplete', 'declined', 'waiting', 'cancelled', 'succeeded', 'deleted', 'refund', 'paymentRequested');
+
+-- CreateEnum
+CREATE TYPE "payout_type" AS ENUM ('cash', 'bank');
+
+-- CreateEnum
+CREATE TYPE "payout_status" AS ENUM ('initial', 'invalid', 'incomplete', 'declined', 'cancelled', 'succeeded');
+
+-- CreateEnum
+CREATE TYPE "account_holder_type" AS ENUM ('individual', 'company');
+
+-- CreateEnum
+CREATE TYPE "bank_account_status" AS ENUM ('new', 'validated', 'verified', 'verification_failed', 'errored');
+
+-- AlterEnum
+BEGIN;
+CREATE TYPE "beneficiary_type_new" AS ENUM ('individual', 'company');
+ALTER TABLE "beneficiaries" ALTER COLUMN "type" TYPE "beneficiary_type_new" USING ("type"::text::"beneficiary_type_new");
+ALTER TYPE "beneficiary_type" RENAME TO "beneficiary_type_old";
+ALTER TYPE "beneficiary_type_new" RENAME TO "beneficiary_type";
+DROP TYPE "beneficiary_type_old";
+COMMIT;
+
+-- AlterEnum
+-- This migration adds more than one value to an enum.
+-- With PostgreSQL versions 11 and earlier, this is not possible
+-- in a single migration. This can be worked around by creating
+-- multiple migrations, each migration adding only one value to
+-- the enum.
+
+
+ALTER TYPE "person_relation" ADD VALUE 'myself';
+ALTER TYPE "person_relation" ADD VALUE 'myorg';
+
+-- DropForeignKey
+ALTER TABLE "beneficiaries" DROP CONSTRAINT "beneficiaries_person_id_fkey";
+
+-- AlterTable
+ALTER TABLE "beneficiaries" ADD COLUMN     "company_id" UUID,
+ALTER COLUMN "person_id" DROP NOT NULL;
+
+-- AlterTable
+ALTER TABLE "campaigns" ADD COLUMN     "vault_id" UUID,
+DROP COLUMN "currency",
+ADD COLUMN     "currency" "currency" DEFAULT E'BGN';
+
+-- AlterTable
+ALTER TABLE "coordinators" ADD COLUMN     "created_at" TIMESTAMPTZ(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+ADD COLUMN     "updated_at" TIMESTAMPTZ(6);
+
+-- AlterTable
+ALTER TABLE "people" ADD COLUMN     "personal_number" TEXT,
+ADD COLUMN     "vault_id" UUID,
+ALTER COLUMN "address" SET DATA TYPE VARCHAR(100);
+
+-- DropEnum
+DROP TYPE "Currency";
+
+-- CreateTable
+CREATE TABLE "companies" (
+    "id" UUID NOT NULL DEFAULT gen_random_uuid(),
+    "company_name" VARCHAR(100) NOT NULL,
+    "companyNumber" TEXT NOT NULL,
+    "legal_person_name" TEXT,
+    "country_code" CITEXT,
+    "city_id" UUID,
+    "created_at" TIMESTAMPTZ(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" TIMESTAMPTZ(6),
+
+    CONSTRAINT "companies_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "benefactors" (
+    "id" UUID NOT NULL DEFAULT gen_random_uuid(),
+    "person_id" UUID NOT NULL,
+    "customer_id" TEXT,
+    "created_at" TIMESTAMPTZ(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" TIMESTAMPTZ(6),
+
+    CONSTRAINT "benefactors_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "vaults" (
```

Comment:
The vault can be either personal or assigned to campaign, so this makes the connection:

- one Person has one Vault
- one Campaign has one Vault

In that scenario the vault record does not store internally whos the person or campaign is, otherwise we'll have

VaultType|Person ID|Campaign ID
---|---|---
Campaign|null|uuid
Person|uuid|null


> 1 campaign to have N vaults

What's the usecase for this one?

Reply:
Hm, after checking the diagram again and the doc from Julian - it looks we don't have a case with personal vaults. 

Then for the relation One Campaign to have one Vault - I find it more natural if we the Vault Table to contain a foreign key to the Campaign table. Then in future if we need to have campaigns with multiple vaults (for whatever reasons) it will not require any database change.

Reply:
Okay, so we remove the relations from `person <> vault` then?

Reply:
I just updated the diagram related to the vaults

Please check it out

![Podkrepi Customer Journey - Money flow - no person vault](https://user-images.githubusercontent.com/893608/135742139-bd15d56d-3703-40a2-8262-c089ac12ace6.jpg)